chrome/content/zotero/elements/contextPane.js

```diff
@@ -0,0 +1,335 @@
+/*
+	***** BEGIN LICENSE BLOCK *****
+	
+	Copyright © 2024 Corporation for Digital Scholarship
+					 Vienna, Virginia, USA
+					 https://www.zotero.org
+	
+	This file is part of Zotero.
+	
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+	
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+	
+	***** END LICENSE BLOCK *****
+*/
+
+
+{
+	class ContextPane extends XULElementBase {
+		content = MozXULElement.parseXULToFragment(`
+			<deck id="zotero-context-pane-deck" flex="1" selectedIndex="0">
+				<deck id="zotero-context-pane-item-deck"></deck>
+				<deck id="zotero-context-pane-notes-deck" class="notes-pane-deck" flex="1"></deck>
+			</deck>
+		`);
+
+		get sidenav() {
+			return this._sidenav;
+		}
+
+		set sidenav(sidenav) {
+			this._sidenav = sidenav;
+			// TODO: decouple sidenav and contextPane
+			sidenav.contextNotesPane = this._notesPaneDeck;
+		}
+
+		get viewType() {
+			return ["item", "notes"][this._panesDeck.getAttribute('selectedIndex')];
+		}
+
+		set viewType(viewType) {
+			let viewTypeMap = {
+				item: "0",
+				notes: "1",
+			};
+			if (!(viewType in viewTypeMap)) {
+				throw new Error(`ContextPane.viewType must be one of ["item", "notes"], but got ${viewType}`);
+			}
+			this._panesDeck.selectedIndex = viewTypeMap[viewType];
+		}
```

Comment:
If we get rid of `viewMode`, I think this can just be `mode` rather than `viewType`. It's really putting the pane into a different mode more than it's changing anything about the view. The deck is just an implementation detail.